[Automatic Docs] Diagram first docementation for pipelex

ivanmilevtues opened the pull request:
### 🔗 Related Issues

### 📝 Description

In this change I've added [CodeBoarding's](https://www.codeboarding.org) free github action, which generates diagram first documentation for python projects. I integrated it with your existing mkdoc structure.

![pipelex](https://github.com/user-attachments/assets/b5bf910f-070a-4a0b-b118-d731d3ee38f5)

### 🔄 Type of Change
- [ ] 🐛 Bug fix
- [ ] ✨ New feature
- [ ] 💥 Breaking change
- [x] 📚 Documentation update
- [ ] 🧹 Code refactor
- [ ] ⚡ Performance improvement
- [ ] ✅ Test update

### 🧪 Tests

### 📋 Checklist

- [ ] Linting / type-checking / unit tests pass locally with my changes
- [ ] I've added tests that prove my fix is effective or that my feature works
- [ ] My code follows the style guidelines of this project
- [x] I've made corresponding changes to the documentation
- [ ] My changes generate no new warnings

ivanmilevtues committed:
Added workflows.

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 8
- JSON files created/updated: 9
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/CodeBoarding/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

ivanmilevtues committed:
Added generated docs to the mkdoc

ivanmilevtues committed:
Fixed links to the main repo from fork.

ivanmilevtues commented:
Unsure how to sign the CLA, but considered it signed when I understand how to do that :D

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 8
- JSON files created/updated: 9
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/Pipelex/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 8
- JSON files created/updated: 9
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/Pipelex/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 6
- JSON files created/updated: 7
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/Pipelex/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

Josephrp commented:
hey i was actually going to use pipelex to make a mermaid docs gen agent pipeline for code repos (just using existing pipelex library) would you be interested in working together on that ?

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 7
- JSON files created/updated: 8
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/Pipelex/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

ivanmilevtues commented:
@Josephrp I don't have much time at the moment. But we can do a quick chat and discuss it. Let me know when you'd have time for it, you can directly book some time with me: https://calendly.com/ivan-i-milev/codeboarding-demo

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 8
- JSON files created/updated: 9
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/Pipelex/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

thomashebrard commented:
> @Josephrp I don't have much time at the moment. But we can do a quick chat and discuss it. Let me know when you'd have time for it, you can directly book some time with me: https://calendly.com/ivan-i-milev/codeboarding-demo

Hello guys, @Josephrp, @ivanmilevtues. I would love to discuss CodeBoarding and the applications around it with you.
Would you be open to take some time in a voice channel on the Pipelex [Discord](https://discord.com/invite/PmrvnkqB8r) to talk more about CodeBoarding, and @Josephrp's project?
And we could also discuss the integration of CodeBoarding

ivanmilevtues commented:
@thomashebrard I joined the discord so we can discuss some time. I will follow up there :)

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 6
- JSON files created/updated: 7
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/Pipelex/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 8
- JSON files created/updated: 9
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/Pipelex/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 5
- JSON files created/updated: 6
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/Pipelex/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

actions-user committed:
docs: update codeboarding documentation in advanced-customization

## 📚 Documentation Update
This commit contains updated documentation files fetched from the CodeBoarding service and copied to the advanced-customization section.

### 📊 Summary
- Documentation files created/updated: 3
- JSON files created/updated: 4
- Documentation directory: .codeboarding/
- JSON directory: .codeboarding/
- Output format: .md
- Repository analyzed: https://github.com/Pipelex/pipelex
- Destination: docs/pages/advanced-customization/

The generated .md files have been automatically copied to the advanced-customization documentation section.

🤖 This commit was automatically generated by the CodeBoarding documentation update workflow.

lchoquel commented:
These docs are obsolete, sorry
